### DISCUSSION TITLE
Option to leave out packet TS in IKEv2 negotiations

### DISCUSSION THREAD

**Discussion by mmirold** [strongswan/strongswan#1280] [Ideas]
Dear strongSwan team,

we are currently trying to establish an IPsec connection with a Juniper SRX using IKEv2. As you might know, Juniper is one of those implementations that don't accept multiple traffic selectors in CREATE_CHILD_SA TSr or TSi payloads. As it seems from a cursory glance at the source code, with renegotiations and traffic-initiated child SA negotiations, the packet traffic selector is always added as a second TS. Is this assumption correct? If it's not, I'm sorry for bothering you.

Otherwise, I understand that, according to RFC 5996, section 2.9, there is nothing wrong with strongSwan adding these selectors. However, I wonder if it would make some people's life easier if you provided an option that disables adding those selectors. Currently, our only alternative seems to be going back to IKEv1 or using DPD instead of using a trap-based policy.

Thanks,
Michael

**Comment by tobiasbrunner** [comment]
> As it seems from a cursory glance at the source code, with renegotiations and traffic-initiated child SA negotiations, the packet traffic selector is always added as a second TS. Is this assumption correct?

Yes, the traffic selectors from an acquire are added to the request. However, they are sent as first TS, not second. And they are not sent when renegotiating CHILD_SAs (assuming you mean rekeying), as they are not stored anywhere.

> Otherwise, I understand that, according to RFC 5996, section 2.9

You should refer to RFC 7296.

> However, I wonder if it would make some people's life easier if you provided an option that disables adding those selectors

There already is one: `charon.ignore_acquire_ts` (see [the docs](https://docs.strongswan.org/docs/5.9/config/strongswanConf.html)).

**Comment by mmirold** [comment]
Thanks Tobias :-) And sorry. I thought I had checked everywhere.
